### DISCUSSION TITLE
Open webots through terminal can detected ros,while open webots through icons dosen't work

### DISCUSSION THREAD

**Discussion by HuNingHe** [cyberbotics/webots#4381] [Q&A]
on ubuntu 20.04 ,
roscore first
then open webots by click icons.
then change the controller to ros
then get the error:
/usr/local/webots/projects/default/controllers/ros/ros: error while loading shared libraries: libroscpp.so: cannot open shared object file: No such file or directory.

while open webots by terminal, it works.
why? and how to fix this?

**Comment by omichel** [comment]
I guess this is because your desktop environment misses some ROS setup. When you install ROS, some stuff is added to your `.bashrc` or `.bash_profile` configuration file that will setup some environment variable when a terminal with bash is started. This is not executed when you launch Webots from the desktop. A solution would be to set these environment variables in the `.profile` file which is executed when the desktop is started. More detail on how to setup environment variable on Ubuntu are provided here: https://help.ubuntu.com/community/EnvironmentVariables. Please let me know if that helps.

**Reply by HuNingHe** [reply]
No it doesn't work

**Reply by HuNingHe** [reply]
I will use terminal to setup webots for now
